### PR TITLE
fmu v5 features - against PX4 older nuttx. 

### PR DESCRIPTION
**Pull request by davids5**
The first 10 comment are cherry picked from https://github.com/PX4/Firmware/pull/7866

The last 3 add board level control of polarity, drive for the UI and status LED on FMUv5
@simonegu There is also interim fix for swapped R and G lines on RC00 test HW.

### CODE REVIEW

File: src/drivers/boards/px4fmu-v5/px4fmu_timer_config.c [hunk 1]
```diff
@@ -193,41 +193,98 @@ __EXPORT const struct io_timers_t led_pwm_timers[MAX_LED_TIMERS] = {
...
+ * if the LED has a 5 Volt supply that the LED will be
+ * off when high.
+ */
+#define CCER_C1_NUM_BITS   4
+#define ACTIVE_LOW(c)      (GTIM_CCER_CC1P << (((c)-1) * CCER_C1_NUM_BITS))
+#define ACTIVE_HIGH(c)     0
+
+#if defined(BOARD_LED_PWM_DRIVE_ACTIVE_LOW)
+#  define POLARITY(c)      ACTIVE_LOW(c)
+#  define DRVIE_TYPE(p)    ((p)|GPIO_OPENDRAIN)
```
Comment (LorenzMeier) [inline src/drivers/boards/px4fmu-v5/px4fmu_timer_config.c]: Typo - should be "DRIVE_TYPE".

File: src/drivers/boards/px4fmu-v5/px4fmu_timer_config.c [hunk 2]
```diff
@@ -193,41 +193,98 @@ __EXPORT const struct io_timers_t led_pwm_timers[MAX_LED_TIMERS] = {
...
+#  define UI_DRVIE_TYPE(p)  ((p)|GPIO_OPENDRAIN)
+#else
+#  define UI_POLARITY(c)    ACTIVE_HIGH((c))
+#  define UI_DRVIE_TYPE(p)  (p)
+#endif
+
 __EXPORT const struct timer_io_channels_t led_pwm_channels[MAX_TIMER_LED_CHANNELS] = {
 	{
-		.gpio_out = LED_TIM3_CH4OUT,
+		.gpio_out = DRVIE_TYPE(LED_TIM3_CH4OUT),
```
Comment (LorenzMeier) [inline src/drivers/boards/px4fmu-v5/px4fmu_timer_config.c]: Resulting from first typo

File: src/drivers/boards/px4fmu-v5/px4fmu_timer_config.c [hunk 3]
```diff
@@ -193,41 +193,98 @@ __EXPORT const struct io_timers_t led_pwm_timers[MAX_LED_TIMERS] = {
...
 		.gpio_in  = 0,
 		.timer_index = 0,
 		.timer_channel = 4,
+		.masks = POLARITY(4),
 	},
 	{
-		.gpio_out = LED_TIM3_CH1OUT,
+		.gpio_out = DRVIE_TYPE(LED_TIM3_CH1OUT),
```
Comment (LorenzMeier) [inline src/drivers/boards/px4fmu-v5/px4fmu_timer_config.c]: Same as above, continues below.

**Comment by davids5** [comment]
@LorenzMeier - thank you for catching that. Typo is fixed, rebased on master and force pushed.

**Pull request merged**